### DISCUSSION TITLE
安装依赖死活安装不上

### DISCUSSION THREAD

**Discussion by BuTingBa** [vbenjs/vue-vben-admin#4615] [Q&A]
按照文档内容，一次输入corepack enable、pnpm install后，提示无法下载pnpm 8.12.1：Error: Error when performing the request to https://registry.npmjs.org/pnpm/-/pnpm-9.12.1.tgz; for troubleshooting help, see https://github.com/nodejs/corepack#troubleshooting.

我修改为淘宝源也无济于事。

我的nodejs版本和git版本都是符合文档的，Node.js v20.18.0。pnmp：9.12.1，
也试过用管理员方式运行，也不行。

能否帮帮我？

![image](https://github.com/user-attachments/assets/fe8da340-d2ae-40ec-ad78-208bc88d63a3)

**Comment by ArthurDarkstone** [comment]
根据图片中的内容`UND_ERR_CONNECT_TIMEOUT` 可以推断出是由网络原因导致的。

**Comment by ArthurDarkstone** [comment]
此处网络原因异常，并非修改pnpm 镜像源可以解决。因为此时应用了`corepack enable` ，在输入`pnpm install` 时，会首先下载 pnpm 的最新版本。此处报错显示的`UND_ERR_CONNECT_TIMEOUT ` 是下载pnpm 引起，而不是下载依赖导致。 解决方式：提前安装一个新版本pnpm 即可

**Comment by elarbor** [comment]
只能用pnpm吗？npm行不？

**Reply by ArthurDarkstone** [reply]
> 只能用pnpm吗？npm行不？

是的，详见[https://doc.vben.pro/guide/introduction/quick-start.html#%E5%AE%89%E8%A3%85%E4%BE%9D%E8%B5%96](https://doc.vben.pro/guide/introduction/quick-start.html#%E5%AE%89%E8%A3%85%E4%BE%9D%E8%B5%96)

**Reply by elarbor** [reply]
哦哦，对应的后端项目在哪啊？

**Reply by LapinBambi** [reply]
请问你找到了吗姐妹？

**Comment by DesignHhuang** [comment]
在网络设置中把 ipv6 网络关掉，pnpm 有时回去通过 ipv6 网络请求，如果你不用 ipv6 的话可以直接关掉。
![image](https://github.com/user-attachments/assets/bd522851-661b-4121-a46d-c8bab0c406ff)

**Comment by dingdayu** [comment]
你可能需要的是 ：COREPACK_NPM_REGISTRY: https://registry.npmmirror.com

**Reply by LapinBambi** [reply]
问题解决了，谢谢！
